kernel/src/main/scala/cats/kernel/Comparison.scala

```diff
@@ -0,0 +1,19 @@
+package cats.kernel
+
+/** ADT encoding the possible results of a comparison */
+sealed abstract class Comparison extends Product with Serializable {
+  /** The signum of this comparison */
+  def toInt: Int = this match {
+    case Comparison.GreaterThan => 1
+    case Comparison.EqualTo     => 0
+    case Comparison.LessThan    => -1
+  }
+}
+
+object Comparison {
+  final case object GreaterThan extends Comparison
+  final case object EqualTo     extends Comparison
+  final case object LessThan    extends Comparison
+
+  implicit val catsKernelEqForComparison: Eq[Comparison] = Eq.fromUniversalEquals
```

Comment:
We could probably make this an `Order` as opposed to just an `Eq`. I don't know if it would ever really be used that way in practice, though.